var/spack/repos/builtin/packages/mapl/package.py

```diff
@@ -138,27 +182,38 @@ class Mapl(CMakePackage):
     depends_on("netcdf-c")
     depends_on("netcdf-fortran")
     depends_on("parallel-netcdf", when="+pnetcdf")
-    depends_on("esmf@8.4:", when="@2.34:")
-    depends_on("esmf@8.3:", when="@2.22:")
+    depends_on("esmf@8.5:", when="@2.40:")
+    depends_on("esmf@8.4.1:8.4", when="@2.34:2.39")
+    depends_on("esmf@8.3.0:8.3", when="@2.22:2.33")
```

Comment:
I think you should be able to do, e.g., `esmf@8.4` and that will pick up anything in the 8.4 range.

Reply:
@mathomp4 Can you look at Alex's suggestion, please? I think that's the last thing.

Reply:
Well huh. I *know* I did that! I guess I never pushed. One moment...

Reply:
Tested and pushed.